### PR TITLE
Allow Event-based Input Skim with `das-up-to-nevents.py`

### PR DESCRIPTION
**Pull request by AdrianoDee**
With https://github.com/cms-sw/cmssw/pull/45055 we can run data wfs skimming the input based on a maximum number of events (and relying on the golden data certification json). This PR proposes a way to allow the RelVal machine to use this new feature when the `InputInfo` includes a limit on `events` (and no limit on lumisections or runs).

### CODE REVIEW

**Comment by AdrianoDee** [comment]
Are the checks failures related to the PR? It doesn't seems to me

**Comment by AdrianoDee** [comment]
Hi Geovanny, I've teste with my local setup (using your instructions) and now everything works as it should.

File: core/model/relval_step.py [hunk 1]
```diff
@@ -92,7 +93,8 @@ class RelValStep(ModelBase):
         },
         '_input': {
             'dataset': lambda ds: not ds or ModelBase.lambda_check('dataset')(ds),
-            'label': lambda l: not l or ModelBase.lambda_check('label')(l)
+            'label': lambda l: not l or ModelBase.lambda_check('label')(l),
+            'events': lambda m: m == '' or int(m) > 0,
```
Comment (ggonzr) [inline core/model/relval_step.py:97]:
```suggestion
            'events': lambda m: m == '' or (m.isnumeric() and int(m) > 0),
```

Comment (ggonzr) [reply core/model/relval_step.py:97]: Just to be sure the field is always filled with a numeric value and avoid parsing exceptions

**Review by ggonzr — CHANGES_REQUESTED** [review]
This patch works as expected. I created this [RelVal](https://cms-pdmv-dev.web.cern.ch/relval/relvals?prepid=CMSSW_14_1_0_pre7__TEST_EVENT_BASED_INPUT_fullsim_noPU_2022_14TeV-RunJetMET02024B_500k-00002&shown=1023&page=0&limit=50) and the new script is loaded in the [cmsDriver.py](https://cms-pdmv-dev.web.cern.ch/relval/api/relvals/get_cmsdriver/CMSSW_14_1_0_pre7__TEST_EVENT_BASED_INPUT_fullsim_noPU_2022_14TeV-RunJetMET02024B_500k-00002) command as required. I suggest a simple change to avoid potential exceptions

File: core/model/relval_step.py (hunk 1 above)
Comment (AdrianoDee) [reply core/model/relval_step.py]: Sure, committed :)

**Pull request merged**